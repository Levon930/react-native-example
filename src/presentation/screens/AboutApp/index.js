import React from 'react';
import { StatusBar, ScrollView } from 'react-native';
import { SafeAreaLayout, Header, Text } from 'components';
import { LocalizationContext } from 'services';
import * as Colors from 'presentation/styles/colors';
import styles from './styles';

export class AboutAppScreen extends React.Component {
  static contextType = LocalizationContext;

  render() {
    const { translations } = this.context;
    return (
      <SafeAreaLayout style={styles.container}>
        <StatusBar backgroundColor={Colors.white} barStyle="dark-content" />
        <Header
          title={translations.settings.aboutApp.title}
          withBackButton
          onBackPress={this.props.navigation.goBack}
        />
        <ScrollView>
          <Text style={styles.contentText}>
            {translations.settings.aboutApp.content}
          </Text>
        </ScrollView>
      </SafeAreaLayout>
    );
  }
}
