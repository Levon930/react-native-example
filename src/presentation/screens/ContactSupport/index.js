import React from 'react';
import { observer } from 'mobx-react';
import { View, StatusBar, TouchableWithoutFeedback, Keyboard } from 'react-native';
import { SafeAreaLayout, TextInput, Header } from 'components';
import { LocalizationContext } from 'services';
import * as Colors from 'presentation/styles/colors';
import { AppRoutes } from 'enums';
import { sendMessageUnactive, sendMessageActive } from 'assets';
import Toast from 'react-native-easy-toast';
import { getErrorMessage } from 'utils';
import ViewModel from './viewModel';
import styles from './styles';
@observer
export class ContactSupportScreen extends React.Component {
  static contextType = LocalizationContext;

  viewModel = new ViewModel(this);

  sendMessage = async () => {
    const { translations } = this.context;

    await this.viewModel.validateEmail();
    if (!this.viewModel.emailValidation) {
      this.refs.emailTextInput.trigerValidation();
      this.refs.toast.show(translations.errors.validEmail, 1500);
    } else if (!this.viewModel.validateMessage) {
      this.refs.toast.show(translations.errors.validMessage, 1500);
    } else {
      this.viewModel.contactSupport()
        .then(() => this.props.navigation.navigate(AppRoutes.CONTACT_SUPPORT_SUCCESS))
        .catch((e) => {
          this.refs.toast.show(getErrorMessage(e, translations), 1500);
        });
    }
  }

  render() {
    const { translations } = this.context;
    return (
      <SafeAreaLayout style={styles.safeArea}>
        <StatusBar backgroundColor={Colors.white} barStyle="dark-content" />
        <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
          <View style={styles.mainContainer}>
            <Header 
              title={translations.settings.contactUs.title}
              withBackButton
              onBackPress={this.props.navigation.goBack}
              rightIcon={this.viewModel.disabledButton ? sendMessageUnactive : sendMessageActive}
              rightIconDisabled={this.viewModel.disabledButton}
              onRightIconPress={this.sendMessage}
            />
            <TextInput
              ref="emailTextInput" 
              placeholder={translations.settings.contactUs.emailTextInput}
              autoCompleteType='email'
              autoCapitalize='none'
              keyboardType='email-address'
              value={this.viewModel.email}
              onChangeText={this.viewModel.setEmail}
            />
            <TextInput 
              placeholder={translations.settings.contactUs.messageTextInput} 
              style={styles.messageTextInput}
              multiline
              value={this.viewModel.message}
              onChangeText={this.viewModel.setMessage}
            />
            <View style={styles.footer} /> 
          </View>
        </TouchableWithoutFeedback>
        <Toast 
          ref="toast" 
          position="center"
        />
      </SafeAreaLayout>
    );
  }
}
