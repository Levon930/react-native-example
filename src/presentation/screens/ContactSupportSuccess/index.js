import React from 'react';
import { View, Image } from 'react-native';
import { SafeAreaLayout, Button, Text } from 'components';
import { LocalizationContext } from 'services';
import { bigCheckedIcon } from 'assets';
import styles from './styles';

export class ContactSupportSuccessScreen extends React.Component {
  static contextType = LocalizationContext;

  render() {
    const { translations } = this.context;
    return (
      <SafeAreaLayout style={styles.safeArea}>
        <View style={styles.mainContainer}>
          <View style={styles.contentContainer} >
            <Image 
              source={bigCheckedIcon}
            />
            <Text style={styles.textMargins}>
              {translations.settings.contactUs.messageSend}
            </Text>
            <Text>
              {translations.general.thanks}
            </Text>
          </View>
          <Button 
            title={translations.settings.contactUs.super}
            style={styles.cancelButton}
            filled
            onPress={() => this.props.navigation.pop(2)}
          />
        </View>
      </SafeAreaLayout>
    );
  }
}
