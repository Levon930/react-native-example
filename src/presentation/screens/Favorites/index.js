import React from 'react';
import { observer } from 'mobx-react';
import { View, StatusBar, FlatList, RefreshControl } from 'react-native';
import { LocalizationContext } from 'services';
import * as Colors from 'presentation/styles/colors';
import Toast from 'react-native-easy-toast';
import { FavoriteTypes, AppRoutes } from 'enums';
import { getName, getRouteAbbrev } from 'utils';
import { StopsStore, AllRoutesStore } from 'stores';
import { Row, Divider, RouteListItem, Text } from '../../components';
import { TabItem, StopsListItem } from './components';
import ViewModel from './viewModel';
import styles from './styles';
@observer
export class FavoritesScreen extends React.Component {
  static contextType = LocalizationContext;

  componentDidMount() {
    this._unsubscribe = this.props.navigation.addListener('focus', () => {
      this.forceUpdate();
    });
  }

  componentDidUpdate() {
    const { translations } = this.context;
    if (this.viewModel.callToast === true) {
      this.refs.toast.show(
        translations.vehicle.toastCannotTrackMore,
        2000
      );
      this.viewModel.toggleToast();
    }
  }

  componentWillUnmount() {
    this._unsubscribe();
  }

  viewModel = new ViewModel(this);

  render() {
    const { translations, appLanguage } = this.context;
    return (
      <View style={styles.safeArea}>
        <StatusBar translucent backgroundColor={Colors.black} barStyle="light-content" />
        <Row style={styles.tabsContainer}>
          <TabItem
            text={translations.favorites.routesTab}
            focused={this.viewModel.activeTab === FavoriteTypes.VEHICLES}
            onPress={() => this.viewModel.setActiveTab(FavoriteTypes.VEHICLES)}
          />
          <TabItem
            text={translations.favorites.stopsTab}
            focused={this.viewModel.activeTab === FavoriteTypes.STOPS}
            onPress={() => this.viewModel.setActiveTab(FavoriteTypes.STOPS)}
          />
        </Row>
        <Choose>
          <When condition={this.viewModel.activeTab === FavoriteTypes.STOPS}>
            <If condition={this.viewModel.stops.length === 0}>
              <Text style={styles.emptyList}>{translations.emptyLists.favoriteStops}</Text>
            </If>
            <FlatList
              data={this.viewModel.stops}
              scrollEventThrottle={20}
              refreshControl={
                <RefreshControl
                  refreshing={this.viewModel.refreshing}
                  onRefresh={this.viewModel.refreshStops}
                />
              }
              renderItem={({ item }) =>
                <StopsListItem
                  name={getName(item, appLanguage)}
                  key={item.id}
                  item={item}
                  onSetFavorite={() => { StopsStore.onSetFavoriteStop(item); this.forceUpdate(); }}
                  isFavorite={StopsStore.isFavorite(item.id)}
                  onPress={() => {
                    this.props.navigation.navigate(AppRoutes.HOME, {
                      stopId: item.id,
                    });
                  }}
                  appLanguage={appLanguage}
                />
              }
              ItemSeparatorComponent={() => (<Divider />)}
            />
          </When>
          <Otherwise>
            <If condition={this.viewModel.routes.length === 0}>
              <Text style={styles.emptyList}>{translations.emptyLists.favoriteRoutes}</Text>
            </If>
            <FlatList
              data={this.viewModel.routes}
              scrollEventThrottle={20}
              refreshControl={
                <RefreshControl
                  refreshing={this.viewModel.refreshing}
                  onRefresh={this.viewModel.refreshRoutes}
                />
              }
              renderItem={({ item }) =>
                <RouteListItem
                  onItemPress={() => {
                    this.props.navigation.navigate(AppRoutes.CHOSEN_ROUTE, {
                      routeItem: item,
                      vehAbbrev: getRouteAbbrev(item.vehicleType, appLanguage),
                      forwardFrom: item.forward.stops.length > 0 && getName(item.forward.stops[0], appLanguage),
                      forwardTo: item.forward.stops.length > 0
                        && getName(item.forward.stops[item.forward.stops.length - 1], appLanguage),
                    });
                  }}
                  type={item.type}
                  forwardFrom={item.forward.stops.length > 0 && getName(item.forward.stops[0], appLanguage)}
                  forwardTo={
                    item.forward.stops.length > 0 &&
                    getName(item.forward.stops[item.forward.stops.length - 1], appLanguage)
                  }
                  vehAbbrev={getRouteAbbrev(item.type, appLanguage)}
                  number={item.number && item.number.trim()}
                  key={item.id}
                  onSubscribe={() => { this.viewModel.onSubscribe(item); this.forceUpdate(); }}
                  onSetFavorite={() => {
                    this.viewModel.onSetFavoriteRoute(item);
                    this.forceUpdate();
                  }}
                  isFollowed={AllRoutesStore.isRouteSubscribed(item.id)}
                  isFavorite={AllRoutesStore.isRouteFavorite(item.id)}
                />
              }
              ItemSeparatorComponent={() => (<Divider />)}
            />
          </Otherwise>
        </Choose>
        <Toast
          ref='toast'
          fadeInDuration={350}
          fadeOutDuration={1000}
          opacity={0.8}
        />
      </View>
    );
  }
}
