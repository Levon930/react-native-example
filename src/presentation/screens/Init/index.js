import React from 'react';
import { AppRoutes } from 'enums';
import { resetInitRouteAndNavigate } from 'navigation';
import { LocalizationContext } from 'services';
import { observer } from 'mobx-react';
import { LoaderView } from 'components';
import { StopsStore, AllRoutesStore } from '../../../stores';
import ViewModel from './viewModel';

@observer
export class InitScreen extends React.Component {
  static contextType = LocalizationContext;

  constructor(props) {
    super(props);
    StopsStore.setStops();
    AllRoutesStore.setRoutesIsVisibleSuburban();
  }

  async componentDidMount() {
    await this.context.initializeAppLanguage();
    this.viewModel.setLaunchStatus(this.computeRoute, this.context.appLanguage);
  }

  viewModel = new ViewModel(this);

  computeRoute = () => {
    const routeName = this.viewModel.isFirstLaunch === (true).toString()
      ? AppRoutes.ONBOARDING : AppRoutes.HOME;
    setTimeout(() => resetInitRouteAndNavigate(routeName, this.props.navigation, null), 3000);
  }

  render() {
    const { initializeAppLanguage } = this.context;
    initializeAppLanguage();
    return (
      <LoaderView />
    );
  }
}
