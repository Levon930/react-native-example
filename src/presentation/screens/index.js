export { OnboardingScreen } from './Onboarding';
export { HomeScreen } from './Home';
export { VehiclesScreen } from './Vehicles';
export { TicketsScreen } from './Tickets';
export { FavoritesScreen } from './Favorites';
export { SettingsScreen } from './Settings';
export { ContactSupportScreen } from './ContactSupport';
export { ContactSupportSuccessScreen } from './ContactSupportSuccess';
export { LanguagesScreen } from './Languages';
export { NotificationsScreen } from './Notifications';
export { UserAgreementScreen } from './UserAgreement';
export { AboutAppScreen } from './AboutApp';
export { ChosenRouteScreen } from './ChosenRoute';
export { InitScreen } from './Init';
export { RouteDetailsScreen } from './RouteDetails';
export { TimeFormatsScreen } from './TimeFormats';
