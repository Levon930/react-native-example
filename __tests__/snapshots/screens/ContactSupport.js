import React from 'react';
import renderer from 'react-test-renderer';
import { ContactSupportScreen } from 'screens';

const navigation = {
  goBack: () => {},
  addListener: () => {},
  navigate: () => {},
};

it('render ContactSupport screen correctly', () => {
  const tree = renderer.create(
    <ContactSupportScreen navigation={navigation} />
  ).toJSON();
  expect(tree).toMatchSnapshot(); 
});
