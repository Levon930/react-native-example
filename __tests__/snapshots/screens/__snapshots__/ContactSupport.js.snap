// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ContactSupport screen correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
      },
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 10,
        "paddingTop": 0,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "marginBottom": 25,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            undefined,
            Object {
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <If
          condition={true}
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginBottom": 25,
                "marginTop": 10,
                "opacity": 1,
              }
            }
          >
            <Image />
          </View>
        </If>
        <If>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginTop": 10,
                "opacity": 1,
              }
            }
          >
            <Image />
          </View>
        </If>
      </View>
      <If
        condition={false}
      >
        <View
          style={
            Object {
              "marginTop": 48,
            }
          }
        />
      </If>
      <Text
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#293043",
                "fontWeight": "500",
              },
              Object {
                "fontSize": 24,
              },
            ],
          ]
        }
      >
        Contact support
      </Text>
    </View>
    <View>
      <If
        condition=""
      >
        <Text
          caption={true}
          style={
            Array [
              Object {
                "fontSize": 15,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "fontSize": 12,
                },
                Object {
                  "color": "#b3b3b3",
                },
              ],
            ]
          }
        >
          Your Email
        </Text>
      </If>
      <View
        centered={true}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "alignItems": "center",
            },
            undefined,
          ]
        }
      >
        <If>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 15,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          />
        </If>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCompleteType="email"
          keyboardType="email-address"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Your Email"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "flex": 1,
                "minHeight": 40,
              },
              null,
              undefined,
              Object {
                "color": "black",
              },
            ]
          }
          underlineColorAndroid="#0000"
        />
      </View>
      <If>
        <View
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              undefined,
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            focusable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
          />
        </View>
      </If>
      <If
        condition={true}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#e5e5e5",
              },
              Object {
                "height": 0.5,
              },
            ]
          }
        />
      </If>
      <If>
        <Text
          caption={true}
          secondary={true}
          style={
            Array [
              Object {
                "fontSize": 15,
              },
              undefined,
              Object {
                "color": "#B7C3CF",
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "marginTop": 8,
              },
            ]
          }
        />
      </If>
      <If>
        <View
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              undefined,
              Object {
                "alignItems": "flex-end",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "height": 30,
                "justifyContent": "center",
                "opacity": 1,
                "width": 30,
              }
            }
          >
            <Image
              style={
                Array [
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </If>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "marginTop": 10,
        }
      }
    >
      <If
        condition=""
      >
        <Text
          caption={true}
          style={
            Array [
              Object {
                "fontSize": 15,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "fontSize": 12,
                },
                Object {
                  "color": "#b3b3b3",
                },
              ],
            ]
          }
        >
          Message
        </Text>
      </If>
      <View
        centered={true}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "alignItems": "center",
            },
            undefined,
          ]
        }
      >
        <If>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 15,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          />
        </If>
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Message"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "flex": 1,
                "minHeight": 40,
              },
              Object {
                "paddingTop": 12,
              },
              undefined,
              Object {
                "color": "black",
              },
            ]
          }
          underlineColorAndroid="#0000"
          value=""
        />
      </View>
      <If>
        <View
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              undefined,
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            focusable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
          />
        </View>
      </If>
      <If
        condition={true}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#e5e5e5",
              },
              Object {
                "height": 0.5,
              },
            ]
          }
        />
      </If>
      <If>
        <Text
          caption={true}
          secondary={true}
          style={
            Array [
              Object {
                "fontSize": 15,
              },
              undefined,
              Object {
                "color": "#B7C3CF",
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "marginTop": 8,
              },
            ]
          }
        />
      </If>
      <If>
        <View
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              undefined,
              Object {
                "alignItems": "flex-end",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "height": 30,
                "justifyContent": "center",
                "opacity": 1,
                "width": 30,
              }
            }
          >
            <Image
              style={
                Array [
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </If>
    </View>
    <View
      style={
        Object {
          "height": "40%",
        }
      }
    />
  </View>
</View>
`;
